test/apify/enqueue_links/enqueue_links.test.ts

```diff
@@ -356,9 +366,9 @@ describe('enqueueLinks()', () => {
             };
 
             const pseudoUrls = [
-                new PseudoUrl('https://example.com/[(\\w|-|/)*]', { method: 'POST' }),
+                new PseudoUrl('https://example.com/[(\\w|-|/)*]'),
                 null,
-                new PseudoUrl('[http|https]://cool.com/', { userData: { foo: 'bar' } }),
+                new PseudoUrl('[http|https]://cool.com/'),
```

Comment:
same here. 

i would also expect the `null` to actually trigger validation, i guess that was some kind of hack in case parsing failed, but ignoring that seems like a wrong thing to do

Reply:
Not sure what you mean here. If there's just `null` (i.e. no patterns provided) - it will mean that all links found on page will be enqueued. If there are some patterns in an array and one of them is `null` - it kinda goes against each other (enqueue matching vs enqueue all) - so it seems reasonable to me that it should throw. Or you meant something else?

Otherwise ready for review.

Reply:
I dont understand why we should even allow the `null` as a valid input, thats what I mean. Imho `null` should trigger validation = be disallowed. I dont see a single reason why it would be good idea to keep it, given we now have enqueing strategies that are more explicit. 

And the "same here" was about using `PseudoUrl` constructors, that's what i said above is not necessary and should be removed from supported input too - currently there is no benefit in this, the class does not do anything else than buiding the regexp and testing it.

On the other hand the test was originally using custom `userData` and the one before was having `POST` method, both of which should be still supported - but no longer via the `PseudoUrl` class. we should still test this.

Reply:
Adjusted the `null` option :+1:

Rest (`PseudoUrl` constructor) was already removed from everywhere.

Reply:
And yeah - tests were also adjusted.